edb/schema/sources.py

```diff
@@ -74,7 +74,7 @@ def populate_pointer_set_for_source_union(
     components: Iterable[Source],
     union: Source,
     *,
-    modname: str = '__derived__',
+    modname: Optional[str] = '__derived__',
```

Comment:
This change doesn't look correct. `modname` is not `Optional` in this function (which is why it's given a default).

Reply:
The ambiguity that caused doubt in me, is that the `get_or_create_union_pointer` function in `pointers` was already annotated with `modname: Optional[str] = None`. This argument is not altered in `get_or_create_union_pointer`, and it is passed directly to `s_sources.populate_pointer_set_for_source_union`.

Should `get_or_create_union_pointer` be modified to have a default value as well, or  should we assert that `modname` is not None before calling the function in `sources`?

Reply:
OK, the other way to fix this is to make it `Optional`, default to `None` and do a runtime default to `__derived__`. (`if modname is None: modname = '__derived__'`)